### ISSUE TITLE
False positive in object_usage_linter when naming non-generic S3 method arguments

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1932]
```r
lintr::lint(
  text = "foo <- function(x) {\n  round(x, units = 'days')\n}",
  linters = lintr::object_usage_linter()
)
<text>:2:3: warning: [object_usage_linter] unused argument (units = "days"). Disable this with a comment: `# nolint` (one line) or `# lintr: disable` (multiple lines). See go/tidy-style#linting for more.
  round(x, units = 'days')
  ^~~~~~~~~~~~~~~~~~~~~~~~
```

This is due to a bug in codetools; filing here in case we think it warrants a workaround on our side:

https://gitlab.com/luke-tierney/codetools/-/issues/12
